### PR TITLE
Added onChanged property to TextFormField

### LINKED ISSUE #28914: Add onChanged to TextFormField
`TextFormField` doesn't have the `onChanged` that `TextField` has, using a `TextEditingController` can be much more complicated, specially when using the BLoC pattern, so adding a `onChanged` callback could simplify this a lot.

### PR DESCRIPTION
**Pull request by kasper9n**
## Description

Adds `onChanged` property to the `TextFormField` material widget.

## Related Issues

Resolves #28914

## Tests

I added the following tests:

- `onChanged callbacks are called` in `text_form_field_test.dart`.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

*No documentation changes were needed as it's already covered.

## Breaking Change

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/material/text_form_field.dart [hunk 1]
```diff
@@ -165,7 +172,8 @@ class TextFormField extends FormField<String> {
         minLines: minLines,
         expands: expands,
         maxLength: maxLength,
-        onChanged: field.didChange,
+        // onChanged: field.didChange,
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/text_form_field.dart]: Remove this line?

File: packages/flutter/test/material/text_form_field_test.dart [hunk 2]
```diff
@@ -123,6 +123,28 @@ void main() {
...
+              },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.enterText(find.byType(TextField), 'Soup');
+    await tester.pump();
+    expect(_called, true);
```
Comment (LongCatIsLooong) [inline packages/flutter/test/material/text_form_field_test.dart]: Could you also verify that the `onChanged` callback is called with the correct value (in this case, `'Soup'`)?

**Review by LongCatIsLooong — COMMENTED** [review]
Thanks for the PR! LGTM.

File: packages/flutter/lib/src/material/text_form_field.dart (hunk 1 above)
Comment (SpectralKH) [reply packages/flutter/lib/src/material/text_form_field.dart]: Removed, whoops

File: packages/flutter/test/material/text_form_field_test.dart (hunk 2 above)
Comment (SpectralKH) [reply packages/flutter/test/material/text_form_field_test.dart]: Made it test for that, and it passes.

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. Thanks for the quick response! Do you mind merging from `upstream/master` to pick up the changes from https://github.com/flutter/flutter/pull/35130 to make all the tests green? It's totally optional though.

**Comment by SpectralKH** [comment]
@LongCatIsLooong Great, Is that all?

**Comment by LongCatIsLooong** [comment]
@SpectralKH merged. Thanks again for the PR!

**Pull request merged**